content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -2265,6 +2273,11 @@ else if(action instanceof InteractionAction)
 			pipe.setContentEntity(entity);
 			pipe.setHelperId(iAction.getHelperId());
 			
+			//While editing Citation ,toolSession already has ResourceToolAction.ACTION_PIPE , store it in session before it gets overwritten
+			ResourceToolActionPipe citationActionPipe = (ResourceToolActionPipe) toolSession.getAttribute(ResourceToolAction.ACTION_PIPE);
+			if(citationActionPipe != null){
+				toolSession.setAttribute(CITATION_ACTION_PIPE, citationActionPipe);
```

Comment:
I wouldn't call this `CITATION_ACTION_PIPE` as it could be any pipe, it just happens that in out example it's one from citations.